src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -389,14 +396,50 @@ class ContentOpenWith extends PureComponent<Props, State> {
                             integrations={integrations}
                         />
                     )}
-                    {shouldRenderIntegrationPortal && (
+                    {(shouldRenderLoadingIntegrationPortal ||
```

Comment:
Looks like there's a lot of duplicated markdown inside of here. Can you extract the error rendering portion out to a method?

Also, it looks like it's possible to render two of the same `IntegrationPortals` if `shouldRenderErrorIntegrationPortal` is true. Is this intended?

Reply:
It looks like the old markup was accidentally left in the changeset?

Reply:
forgot to delete that second one. It should not be there

Reply:
@JustinHoldstock does this seem more reasonable?

Reply:
Almost! Is there anyway you can reduce the duplicate error markdown?

Reply:
@JustinHoldstock, can you clarify what duplication you mean? It looks pretty DRY to me.

Reply:
@jstoffan @jeremypress Here, and then directly below here are two identical pieces of markdown for an `ErrorMask` element. I'd extract that to a function (`getOpenWithError()`).

```
// First instance as follows
                                {shouldRenderErrorIntegrationPortal ? (
                                    <ErrorMask
                                        errorHeader={
                                            <FormattedMessage
                                                {...messages.executeIntegrationOpenWithErrorHeader}
                                            />
                                        }
                                        errorSubHeader={
                                            <FormattedMessage
                                                {...messages.executeIntegrationOpenWithErrorSubHeader}
                                            />
                                        }
                                    />
                                ) : (
                                    <LoadingIndicator
                                        className="bcow-portal-loading-indicator"
                                        size="large"
                                    />
                                )}

// then below that, but inside an integration portal
                    {shouldRenderErrorIntegrationPortal && (
                        <IntegrationPortal
                            integrationWindow={this.integrationWindow}
                        >
                            <div className="be bcow bcow-portal-container">
                                <ErrorMask
                                    errorHeader={
                                        <FormattedMessage
                                            {...messages.executeIntegrationOpenWithErrorHeader}
                                        />
                                    }
                                    errorSubHeader={
                                        <FormattedMessage
                                            {...messages.executeIntegrationOpenWithErrorSubHeader}
                                        />
                                    }
                                />
                            </div>
                        </IntegrationPortal>
                    )}
```

Both ErrorMasks are the same
```
                                <ErrorMask
                                    errorHeader={
                                        <FormattedMessage
                                            {...messages.executeIntegrationOpenWithErrorHeader}
                                        />
                                    }
                                    errorSubHeader={
                                        <FormattedMessage
                                            {...messages.executeIntegrationOpenWithErrorSubHeader}
                                        />
                                    }
                                />
```

Reply:
Sorry! Looks like either Github or Chrome was showing out of date results, marking as resolved